### PR TITLE
PS-3887 (5.6): TokuDB does not compile with -DWITH_PERFSCHEMA_STORAGE_ENGINE=OFF

### PR DESCRIPTION
**Pull request by inikep**
Add `-DWITHOUT_PERFSCHEMA_STORAGE_ENGINE=ON` to inverted config in Travis config file

Fix TokuDB usage of the following functions to match `psi/mysql_thread.h` definitions:
1. inline_mysql_mutex_lock to `if defined(SAFE_MUTEX) || defined (HAVE_PSI_MUTEX_INTERFACE)`
2. inline_mysql_mutex_unlock to `#ifdef SAFE_MUTEX`

### CODE REVIEW

**Comment by inikep** [comment]
https://jenkins.percona.com/job/percona-server-5.6-param/2129/

**Comment by inikep** [comment]
https://jenkins.percona.com/job/percona-server-5.6-param/2131/

File: storage/innobase/log/log0online.cc [hunk 1]
```diff
@@ -1538,10 +1538,10 @@ log_online_open_bitmap_file_read_only(
 	if (srv_data_home_len
 			&& srv_data_home[srv_data_home_len-1]
 			!= SRV_PATH_SEPARATOR) {
-		ut_snprintf(bitmap_file->name, FN_REFLEN, "%s%c%s",
+		ut_snprintf(bitmap_file->name, sizeof(bitmap_file->name), "%s%c%s",
```
Comment (percona-ysorokin) [inline storage/innobase/log/log0online.cc]: This is a legitimate fix, but it should be applied to 5.5 branch as well. So, please file a new bug and fix this part in all 3 branches.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Review by percona-ysorokin — APPROVED** [review]
LGTM

File: storage/tokudb/tokudb_thread.h [hunk 2]
```diff
@@ -245,15 +245,11 @@ inline void mutex_t::lock(
 #endif
 }
 inline void mutex_t::unlock(
-#ifdef HAVE_PSI_MUTEX_INTERFACE
+#ifdef SAFE_MUTEX
```
Comment (george-lorch) [inline storage/tokudb/tokudb_thread.h]: Why not #if defined(SAFE_MUTEX) like others instead of #ifdef ?

**Review by george-lorch — APPROVED** [review]
OK as is or you can touch up the mentioned #if defined if you wish and merge without further review.

**Comment by peterschwaller** [comment]
touch

**Pull request merged**